fms_fsdp/utils/dataloader_utils.py

```diff
@@ -86,30 +105,15 @@ def causal_lm(data_seq, prompt_len=1):
     # Wrap above dataset in packing logic to form constant-length lines.
     data = Buffer_Dataset(
         data,
-        cfg.seq_length + 1,
+        cfg.seq_length if causal_lm not in postprocess else cfg.seq_length + 1,
         drop_final_token=cfg.sep_token,
         pack_hard=True,
     )
     # Shuffle outputs in length 10k buffer. Consecutive lines appear 10k steps apart on average.
     data = Preload_Buffer_Dataset(data, 10000)
-    # Split line into input and target for the CLM task.
-    data = Preprocess_Dataset(data, causal_lm)
+    # Apply desired postprocessing steps in sequence
+    data = Preprocess_Dataset(data, torch.IntTensor)
```

Comment:
is this just wrapping with IntTensor?

Reply:
Yes - turn list outputs into torch tensors before applying any user-specified preprocess functions